tests/trx_generator/trx_generator.cpp

```diff
@@ -202,4 +205,117 @@ namespace eosio::testing {
       return true;
    }
 
-}
\ No newline at end of file
+   fc::variant json_from_file_or_string(const string& file_or_str, fc::json::parse_type ptype = fc::json::parse_type::legacy_parser)
+   {
+      regex r("^[ \t]*[\{\[]");
+      if ( !regex_search(file_or_str, r) && fc::is_regular_file(file_or_str) ) {
+         try {
+            return fc::json::from_file(file_or_str, ptype);
+         } EOS_RETHROW_EXCEPTIONS(json_parse_exception, "Fail to parse JSON from file: ${file}", ("file", file_or_str));
+
+      } else {
+         try {
+            return fc::json::from_string(file_or_str, ptype);
+         } EOS_RETHROW_EXCEPTIONS(json_parse_exception, "Fail to parse JSON from string: ${string}", ("string", file_or_str));
+      }
+   }
+
+   trx_generator::trx_generator(std::string chain_id_in, const std::string& abi_data_file, std::string contract_owner_account, std::string auth_account, std::string action_name,
+         const std::string& action_data_file_or_str, int64_t trx_expr, const std::string& private_key_str, std::string lib_id_str, std::string log_dir) :
+      _provider(), _chain_id(chain_id_in), _abi_data_file_path(abi_data_file), _contract_owner_account(contract_owner_account), _auth_account(auth_account), _action(action_name), _action_data_file_or_str(action_data_file_or_str),
+      _trx_expiration(trx_expr*1000000), _private_key(fc::crypto::private_key(private_key_str)),
```

Comment:
Would like to see `fc::microseconds` used instead of `int64_t`

Reply:
Addressed: [03750c9](https://github.com/AntelopeIO/leap/pull/626/commits/03750c994aa61d4877ae708c2920245ff8e9526b)